CMakeLists.txt

```diff
@@ -100,7 +100,9 @@ endif()
 
 # --- Linux Joystick API -------------------------------------------------------
 
-check_include_files(linux/joystick.h HAVE_LINUX_JOYSTICK_H)
+if(NOT CORE_SYSTEM_NAME STREQUAL android)
```

Comment:
if(CORE_SYSTEM_NAME STREQUAL linux) is not whitelisting everything else.
See how OSX is treated above, we also explicitly check for OSX there.